### PR TITLE
AAP-5044 - Create AAP Backup and Restore doc

### PR DESCRIPTION
**Pull request by dcdacosta**
This bug addresses the work for https://issues.redhat.com/browse/AAP-5044

Changes in this pull request include:
Creating a new AAP Operator Backup and Restore Guide

Preview link (VPN required): http://file.rdu.redhat.com/~ddacosta/AAP5044/master.html

### CODE REVIEW

File: downstream/modules/platform/con-aap-backup-recommendations.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[id="aap-backup-recommendations"]
+
+= Backup recommendations
+
+
+[role="_abstract"]
+Recovering from data loss requires that you plan for and create backup resources of your {PlatformName} deployments on a regular basis. At a minimum, Red Hat recommends backing up deployments of {PlatformName} under the following circumstances:
+
+* Before upgrading your {PlatormName} deployments
```
Comment (rooftopcellist) [inline downstream/modules/platform/con-aap-backup-recommendations.adoc:9]: It looks like `{PlatormName}` didn't resolve correctly in the preview rendering fyi.  I think it instead needs to be `{PlatformName}`

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+= Backing up the {ControllerNameStart} deployment
+
+[role=_abstract]
+Use this procedure to back up a deployment of the controller, including jobs, inventories, and credentials.
+
+.Prerequisites
+
+* You must be authenticated with an Openshift cluster.
+* The {OperatorPlatform} has been deployed to the cluster.
+* The {PlatformName} is deployed to using the operator.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:12]: This renders as:
```
The Red Hat Ansible Automation Platform is deployed to using the operator.
```

It should instead read:
```
Automation Controller is deployed to using the AAP Operator.
```

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+= Backing up the {ControllerNameStart} deployment
+
+[role=_abstract]
+Use this procedure to back up a deployment of the controller, including jobs, inventories, and credentials.
+
+.Prerequisites
+
+* You must be authenticated with an Openshift cluster.
+* The {OperatorPlatform} has been deployed to the cluster.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:11]: nitpick, this should say installed, not deployed:

```
* The {OperatorPlatform} has been installed to the cluster.
```

(_install_ the operator, _deploy_ the app components like controller and hub)

File: downstream/modules/platform/proc-aap-hub-backup.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+.Procedure
+. Log in to *{OCP}*.
+. Navigate to menu:Operators[Installed Operators].
+. Select the {OperatorPlatform} installed on your project namespace.
+. Select the *Automation Hub Backup* tab.
+. Click btn:[Create AutomationHubBackup].
+. Enter a *Name* for the backup.
+. Enter the *Deployment name* for the deployment being backed up.
+. If you want to use a customer, pre-created pvc:
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-hub-backup.adoc:22]: `custom`, not `customer`

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+. If you want to use a customer, pre-created pvc:
+.. Optionally enter the name of the *Backup persistant volume claim*.
+.. Optionally enter the *Backup PVC storage requirements*, and *Backup PVC storage class*.
++
+[NOTE]
+====
+If no pvc or storage class is provided, the cluster's default storage class is used to create the pvc.
+====
++
+.. If you have a large database, specify higher storage requests under *Backup management pod resource requirements*.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:31]: Could we rephrase this to say:

```
If you have a large database, specify your storage requests accordingly under *Backup management pod resource requirements*.
```

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+[NOTE]
+====
+You can check the size of the existing postgres database data directory by running the following command inside the postgres pod.
+----
+$ df -h | grep "/var/lib/pgsql/data"
+----
+====
+. Click btn:[Create].
++
+A backup tar file of the specified deployment is created and available for data recovery or deployment rollback. Future backups are stored in separate tar files on the same pvc.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:42]: `A backup tar file` --> `A backup tarball`

Maybe this is just tech lingo, but that's how I've heard a tar.gz file referred to.

File: downstream/modules/platform/proc-aap-hub-restore.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
+[id="aap-hub-restore"]
+
+= Recovering the {HubNameStart} deployment
+
+[role=_abstract]
+Use this procedure to restore a previous hub deployment into the namespace. The deployment name you provide will be the name of the new AutomationHub custom resource that will be created.
+
+[NOTE]
+====
+The name specified for the new AutomationHub custom resource must not match an existing deployment or the recovery process will fail.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-hub-restore.adoc:10]: I notice we have this noted twice in this section...  But I'm not wholly against it.  This is an important point I want to make sure customers see.

File: downstream/modules/platform/proc-aap-hub-restore.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+====
+
+.Prerequisites
+
+* You must be authenticated with an Openshift cluster.
+* The {ControllerName} has been deployed to the cluster.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-hub-restore.adoc:16]: This should be `{HubName}` not Controller.

File: downstream/modules/platform/proc-aap-hub-backup.adoc [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+= Backing up the {HubNameStart} deployment
+
+[role=_abstract]
+Use this procedure to back up a deployment of the hub, including all hosted Ansible content.
+
+.Prerequisites
+
+* You must be authenticated with an Openshift cluster.
+* The {OperatorPlatform} has been deployed to the cluster.
+* The {PlatformName} is deployed to using the operator.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-hub-backup.adoc:12]: Same thing here.  This renders as:
```
The Red Hat Ansible Automation Platform is deployed to using the operator.
```

It should instead read:
```
Automation Hub is deployed to using the AAP Operator.
```

File: downstream/modules/platform/proc-aap-controller-restore.adoc [hunk 10]
```diff
@@ -0,0 +1,37 @@
+[id="aap-controller-restore"]
+
+= Recovering the {ControllerNameStart} deployment
+
+[role=_abstract]
+Use this procedure to restore a previous controller deployment from an AutomationControllerBackup. The deployment name you provide will be the name of the new AutomationController custom resource that will be created.
+
+[NOTE]
+====
+The name specified for the new AutomationController custom resource must not match an existing deployment or the recovery process will fail.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-aap-controller-restore.adoc:10]: I notice we have this noted twice in this section... But I'm not wholly against it. This is an important point I want to make sure customers see.

**Review by rooftopcellist — CHANGES_REQUESTED** [review]
There are a few surgical fixes needed that I have detailed.  After that, this PR looks good to me.  Ready for Peer review and merge.

File: downstream/modules/platform/con-aap-backup-recommendations.adoc (hunk 1 above)
Comment (dcdacosta) [reply downstream/modules/platform/con-aap-backup-recommendations.adoc:9]: Fixed.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 2 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:12]: Fixed.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 3 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:11]: Fixed.

File: downstream/modules/platform/proc-aap-hub-backup.adoc (hunk 4 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-hub-backup.adoc:22]: Fixed.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 5 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:31]: Changed.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 6 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:42]: Changed.

File: downstream/modules/platform/proc-aap-hub-restore.adoc (hunk 8 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-hub-restore.adoc:16]: Fixed.

File: downstream/modules/platform/proc-aap-hub-backup.adoc (hunk 9 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-hub-backup.adoc:12]: Fixed.

File: downstream/assemblies/platform/assembly-aap-backup.adoc [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+[id="aap-backup"]
+
+:context: aap-backup
+
+= Creating {PlatformName} backup resources
+
+[role="_abstract"]
+
+Backing up your {PlatformName} deployment involves creating backup resources for your deployed automation hub and automation controller instances. Use these procedures to create backup resources for your {PlatformName} deployment.
```
Comment (jself-sudoku) [inline downstream/assemblies/platform/assembly-aap-backup.adoc:12]: No changes; just replace automation hub and automation controller w/ attributes.

```suggestion
Backing up your {PlatformName} deployment involves creating backup resources for your deployed {HubName} and {ControllerName} instances. Use these procedures to create backup resources for your {PlatformName} deployment.
```

File: downstream/modules/platform/con-aap-backup-recovery.adoc [hunk 12]
```diff
@@ -0,0 +1,18 @@
+[id="aap-backup-recovery"]
+
+= About backup and recovery
+
+[role="_abstract"]
+Red Hat recommends backing up deployments of Red Hat Ansible Automation Platform in your OpenShift Container Platform environment to prevent data loss.
+
+A backup resource of your {PlatformName} deployment includes the following:
+
+* Custom deployment specific values in the spec section of the AAP custom resource object
```
Comment (jself-sudoku) [inline downstream/modules/platform/con-aap-backup-recovery.adoc:10]:
```suggestion
* Custom deployment of specific values in the spec section of the {PlatformNameShort} custom resource object
```

File: downstream/modules/platform/con-aap-backup-recovery.adoc [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+* Back up of the postgresql database
```
Comment (jself-sudoku) [inline downstream/modules/platform/con-aap-backup-recovery.adoc:11]:
```suggestion
* Back up of the `postgresql` database
```

File: downstream/modules/platform/con-aap-backup-recovery.adoc [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+* secret_key, admin_password, and broadcast_websocket secrets,
```
Comment (jself-sudoku) [inline downstream/modules/platform/con-aap-backup-recovery.adoc:12]: Should these all be in ``?
```suggestion
* `secret_key`, `admin_password`, and `broadcast_websocket` secrets,
```

File: downstream/modules/platform/con-aap-backup-recommendations.adoc [hunk 15]
```diff
@@ -0,0 +1,11 @@
+[id="aap-backup-recommendations"]
+
+= Backup recommendations
+
+
+[role="_abstract"]
+Recovering from data loss requires that you plan for and create backup resources of your {PlatformName} deployments on a regular basis. At a minimum, Red Hat recommends backing up deployments of {PlatformName} under the following circumstances:
+
+* Before upgrading your {PlatformName} deployments
+* Before upgrading your Openshift cluster
```
Comment (jself-sudoku) [inline downstream/modules/platform/con-aap-backup-recommendations.adoc:10]: Just a question/thought: Do we use the attribute {OCP} interchangeably with OpenShift? Just wondering about consistency. Wondering if we should have an attribute for just OpenShift. Just a thought since I see OpenShift a few times in addition to the OCPShort attribute.

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 16]
```diff
@@ -0,0 +1,42 @@
+[id="aap-controller-backup"]
+
+= Backing up the {ControllerNameStart} deployment
+
+[role=_abstract]
+Use this procedure to back up a deployment of the controller, including jobs, inventories, and credentials.
+
+.Prerequisites
+
+* You must be authenticated with an Openshift cluster.
```
Comment (jself-sudoku) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:10]: Same question as previous section about OpenShift and whether is should be an attribute. If not, ignore the questions.

File: downstream/modules/platform/proc-aap-controller-backup.adoc [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+* The {OperatorPlatform} has been installed to the cluster.
+* The {ControllerName} is deployed to using the {OperatorPlatform}.
+
+.Procedure
+. Log in to *{OCP}*.
+. Navigate to menu:Operators[Installed Operators].
+. Select the {OperatorPlatform} installed on your project namespace.
+. Select the *Automation Controller Backup* tab.
```
Comment (jself-sudoku) [inline downstream/modules/platform/proc-aap-controller-backup.adoc:18]: Should this one follow the accessibility example:
```suggestion
. Select the btn:[Automation Controller Backup] tab.
```

File: downstream/modules/platform/proc-aap-hub-backup.adoc [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+
+* You must be authenticated with an Openshift cluster.
+* The {OperatorPlatform} has been installed to the cluster.
+* The {HubName} is deployed to using the {OperatorPlatform}.
+
+.Procedure
+. Log in to *{OCP}*.
+. Navigate to menu:Operators[Installed Operators].
+. Select the {OperatorPlatform} installed on your project namespace.
+. Select the *Automation Hub Backup* tab.
```
Comment (jself-sudoku) [inline downstream/modules/platform/proc-aap-hub-backup.adoc:18]: Same question as above in  controller backup section. Should this follow the accessibility tip or is the tip only for buttons or other UI elements?

```suggestion
. Select the btn:[Automation Hub Backup] tab.
```

**Comment by jself-sudoku** [comment]
Just a few updates to attributes and question/comment about accessibility for UI elements. Other than those, LGTM.

File: downstream/modules/platform/con-aap-backup-recommendations.adoc (hunk 15 above)
Comment (dcdacosta) [reply downstream/modules/platform/con-aap-backup-recommendations.adoc:10]: > Just a question/thought: Do we use the attribute {OCP} interchangeably with OpenShift? Just wondering about consistency. Wondering if we should have an attribute for just OpenShift. Just a thought since I see OpenShift a few times in addition to the OCPShort attribute.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 16 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:10]: > Same question as previous section about OpenShift and whether is should be an attribute. If not, ignore the questions.

See above: OCP refers to the OpenShift Container Platform but OpenShift alone, I'm referring to a cluster. I'll ask in the team call whether a generic attribute should be created for the vanilla OpenShift term.

File: downstream/modules/platform/proc-aap-controller-backup.adoc (hunk 17 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-controller-backup.adoc:18]: > Should this one follow the accessibility example or does it only apply to buttons?

No, tabs and other UI elements can use the * bold *. Unfortunately, AsciiDoctor only provides macros for buttons and menus but they should only be used for button and menu items.

File: downstream/modules/platform/proc-aap-hub-backup.adoc (hunk 18 above)
Comment (dcdacosta) [reply downstream/modules/platform/proc-aap-hub-backup.adoc:18]: > Same question as above in controller backup section. Should this follow the accessibility tip or is the tip only for buttons or other UI elements?

No, see response above for explanation.

File: downstream/assemblies/platform/assembly-aap-backup.adoc (hunk 11 above)
Comment (dcdacosta) [reply downstream/assemblies/platform/assembly-aap-backup.adoc]: Good catch!

File: downstream/modules/platform/con-aap-backup-recovery.adoc (hunk 13 above)
Comment (dcdacosta) [reply downstream/modules/platform/con-aap-backup-recovery.adoc]: Fixed.

File: downstream/modules/platform/con-aap-backup-recovery.adoc (hunk 14 above)
Comment (dcdacosta) [reply downstream/modules/platform/con-aap-backup-recovery.adoc]: Fixed.

File: downstream/modules/platform/con-aap-backup-recovery.adoc (hunk 12 above)
Comment (dcdacosta) [reply downstream/modules/platform/con-aap-backup-recovery.adoc]: Fixed.
